mule-module-apikit/src/main/java/org/mule/module/apikit/validation/RestJsonSchemaValidator.java

```diff
@@ -97,11 +105,22 @@ else if (input instanceof byte[])
 
             JsonSchema schema = JsonSchemaCache.getJsonSchemaCache(muleContext, configId, api).get(schemaPath);
             ProcessingReport report = schema.validate(data);
-            if (!report.isSuccess())
+            Iterator<ProcessingMessage> iterator = report.iterator();
+
+            while (iterator.hasNext())
             {
-                String message = report.iterator().hasNext() ? report.iterator().next().toString() : "no message";
-                logger.info("Schema validation failed: " + message);
-                throw new BadRequestException(message);
+                ProcessingMessage next = iterator.next();
+                LogLevel logLevel = next.getLogLevel();
+                String message = next.toString();
+
+                boolean failOnWarning = Boolean.valueOf(
+                        System.getProperty(JSON_SCHEMA_FAIL_ON_WARNING_KEY, "false"));
+
+                if (logLevel.equals(ERROR) || (logLevel.equals(WARNING) && failOnWarning))
+                {
+                    logger.info("Schema validation failed: " + message);
+                    throw new BadRequestException(message);
```

Comment:
Nowadays we are throwing an exception showing only the first error we receive from the report. In this PR we have the same logic, but IMHO this should change, as we should throw all the warnings/errors we know. If you throw an exception when you read the first problem you receive, you could be missing to throw an error. What do you think @svacas ?